pkg/controller/elasticsearch/services/services.go

```diff
@@ -33,29 +32,30 @@ func TransportServiceName(esName string) string {
 // It is used by Elasticsearch nodes to talk to remote cluster nodes.
 func NewTransportService(es esv1.Elasticsearch) *corev1.Service {
 	nsn := k8s.ExtractNamespacedName(&es)
-	return &corev1.Service{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: es.Namespace,
-			Name:      TransportServiceName(es.Name),
-			Labels:    label.NewLabels(nsn),
-		},
-		Spec: corev1.ServiceSpec{
-			Selector: label.NewLabels(nsn),
-			Ports: []corev1.ServicePort{
-				{
-					Protocol: corev1.ProtocolTCP,
-					Port:     network.TransportPort,
-				},
-			},
-			// We set ClusterIP to None in order to let the ES nodes discover all other node IPs at once.
-			ClusterIP:       "None",
-			SessionAffinity: corev1.ServiceAffinityNone,
-			Type:            corev1.ServiceTypeClusterIP,
-			// Nodes need to discover themselves before the pod is considered ready,
-			// otherwise minimum master nodes would never be reached
-			PublishNotReadyAddresses: true,
+	svc := corev1.Service{
+		ObjectMeta: es.Spec.Transport.Service.ObjectMeta,
+		Spec:       es.Spec.Transport.Service.Spec,
+	}
```

Comment:
`svc := es.Spec.Transport.Service.DeepCopy()`?

Reply:
No, because that's a `ServiceTemplate` and we want a `Service` here

Reply:
Got it 👍